json-core/src/main/java/io/micronaut/json/JsonMapper.java

```diff
@@ -228,14 +243,43 @@ default Processor<byte[], JsonNode> createReactiveParser(@NonNull Consumer<Proce
     <T> byte[] writeValueAsBytes(@NonNull Argument<T> type, @Nullable T object) throws IOException;
 
     /**
-     * Write an object as String.
-     *
-     * @param object The object to serialize.
-     * @return The serialized encoded json.
-     * @throws IOException IOException
+     * Write the given value as a string.
+     * @param object The object
+     * @param <T> The generic type
+     * @return The string
+     * @throws IOException If an unrecoverable error occurs
+     */
+    @SuppressWarnings("unchecked")
+    default  @NonNull <T> String writeValueAsString(@NonNull T object) throws IOException {
```

Comment:
this generic param doesn't really make sense, can we move to Object?

Reply:
[changed parameter to Object](https://github.com/micronaut-projects/micronaut-core/pull/9424/commits/9ad2867fb38fd5a6dd346cf954e36fcbb750e4f4)